### ISSUE TITLE
500 Internal Server Error: Cannot find packages

### ISSUE THREAD

**Issue by Stephanevg** [joelspadin-garmin/vscode-private-extension-manager#92]
Hi,

I am using a local instance of artifactory. After quite some troubleshooting to get things to work I i am facing (I believe) the last error before my salvation (🙏🏻).

When I click 'Refresh' in the **extensions** section of the 'Private extension' extension, I get the following error that pop's up:

> 500 Internal Server Error - GET https://bin.woopy.com/artifactory/api/npm/vscodextensions-npm-local/-/v1/search?text=*&size=20&from=0&quality=0.65&popularity=0.98&maintenance=0.5

I have a gut feeling this is because I might have unproperly uploaded my package. (I am a npm noob...just so you know...).

This is what the structure looks like on artifactory:

![image](https://github.com/user-attachments/assets/e06b7aa7-5e53-4782-a0d5-5de8403657bb)

And this what the `package.json` that is contained in the folder looks like this:

```json
//The contributes section can be ignored, as that is suposed to be in the extension itself...
{
  "name": "scw-snippets",
  "displayName": "SCW.Snippets",
  "description": "woopy PowerShell Snippets",
  "contributors": [
    "Stéphane <Stephane@woopy.com>"
  ],
  "Author": "Stéphane van Gulick",
  "publisher": "woopy-Common-Team",
  "version": "0.0.1",
  "engines": {
    "vscode": "^1.95.0"
  },
  "repository": {
    "type": "git",
    "url": "https://tfs.woopy.com/DWF/dwf/_git/script-scripttemplate.git"
  },
  "icon": "/media/woopy.Logo.png",
  "files": [
    "scw-snippets-0.0.1.vsix",
    "media/woopy.Logo.png"
  ],
  "categories": [
    "Snippets"
  ],
  "publishConfig": {
    "registry": "https://bin.woopy.com/artifactory/api/npm/vscodextensions-npm-local"
  },
  "contributes": {
    "snippets": [
      {
        "language": "powershell",
        "path": "./snippets/woopy_Class.code-snippets"
      },
      {
        "language": "powershell",
        "path": "./snippets/woopy_Comment_Based_Help.code-snippets"
      },
      {
        "language": "powershell",
        "path": "./snippets/woopy_Function_Param.code-snippets"
      },
      {
        "language": "powershell",
        "path": "./snippets/woopy_Function_Simple.code-snippets"
      },
      {
        "language": "powershell",
        "path": "./snippets/woopy_Script_Param.code-snippets"
      }
    ],
    "keybindings": [
      {
        "key": "ctrl+shift+y",
        "command": "editor.action.insertSnippet",
        "when": "editorTextFocus",
        "args": {
          "langId": "powershell",
          "name": "woopy_Function_Param"
        }
      },
      {
        "key": "ctrl+shift+a",
        "command": "editor.action.insertSnippet",
        "when": "editorTextFocus",
        "args": {
          "langId": "powershell",
          "name": "woopy_Script_Param"
        }
      }
    ]
  },
  "keywords": [
    "woopy",
    "common",
    "tools"
  ],
  "author": "Stéphane"
}

```

and this is how I uploaded it:

```
cd .\publish
npm publish --registry https://bin.woopy.com/artifactory/api/npm/vscodextensions-npm-local
```

And got the following output when I pushed:

```
PS C:\Users\Stephane\Code\script-scripttemplate\Sources\publish> npm publish --registry https://bin.woopy.com/artifactory/api/npm/vscodextensions-npm-local
npm WARN publish npm auto-corrected some errors in your package.json when publishing.  Please run "npm pkg fix" to address these errors.
npm WARN publish errors corrected:
npm WARN publish Removed invalid "scripts"
npm notice
npm notice 📦  scw-snippets@0.0.1
npm notice === Tarball Contents ===
npm notice 86.4kB media/woopy.Logo.png
npm notice 2.0kB  package.json
npm notice 72.9kB scw-snippets-0.0.1.vsix
npm notice === Tarball Details === 
npm notice name:          scw-snippets
npm notice version:       0.0.1
npm notice filename:      scw-snippets-0.0.1.tgz
npm notice package size:  137.3 kB
npm notice unpacked size: 161.3 kB
npm notice shasum:        1c1c0e7fcd77aca0bb44890a1ba2242420cadb97
npm notice integrity:     sha512-LDDUGnaLYhny7[...]A1BRED2TnS0A==
npm notice total files:   3
npm notice 
npm notice Publishing to https://bin.woopy.com/artifactory/api/npm/vscodextensions-npm-local with tag latest and default access
+ scw-snippets@0.0.1
PS C:\Users\Stephane\Code\script-scripttemplate\Sources\publish> npm publish --registry https://bin.woopy.com/artifactory/api/npm/vscodextensions-npm-local
```

> You will notice it mentions stuff about the `removed invalid scripts` which is strange, as I **don't even had a script tag in package.json file**

Do you guys see anything wrong in the steps I took ?

**Comment by joelspadin-garmin** [comment]
I don't see anything wrong with how you're publishing the package, and the file structure inside the package looks correct. It looks like an issue with Artifactory's handling of the search API instead. I believe the registry I've been testing this against is built with Sonatype Nexus, so I'm unsure what quirks Artifactory might have.

Some things you could try:

1. Access that `.../v1/search/...` URL in a browser. If it works in a browser but not from this extension, then it could be something like the headers sent with the request that causes the error.
2. Create a test program using https://www.npmjs.com/package/libnpmsearch and see if that works. If it does, then either I need to update all of the dependencies here, or there is some option I'm setting that Artifactory doesn't like.
3. If the latest version of libnpmsearch doesn't work, then it seems like either a bug in Artifactory itself or something wrong with your local installation of it. You might need to contact JFrog's support then.

**Comment by Stephanevg** [comment]
Great, thanks for your suggestions. 
I have done some additional tests on my end, and managed to get step closer.. (but not there quite yet... :( )

First of all, executing the query directly in the browser actually worked, but only AFTER I authenticated  (via browser pop up).



I tried everything out in a seperate http call using the below powershell snippet:


```
$UserName = "plop"
$Password= "MyPassword"
$FullCredentials = "$UserName" + ":" + "$Password"
$base64Auth = [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes($FullCredentials ))
$headers = @{Authorization = "Basic $base64Auth"}
$return = Invoke-RestMethod -URI $Url -Headers $headers
```

using my own generated token threw the same error. 

I tried using another already existing account that we use for nuget automation and then my powershell query returned something back.

Looking more closley to it, I generated the wrong token. In Artifactory, there are two type of token:
- Identity tokens
- Api Keys

For some reason, I generated and tried everything using an `identity token` but an `api key` was actually needed. (logic, but for some reason, I didn't saw it immedatley...).

So, I have another issue now, which I think should be tracked in a seperate issue.
